Allow to bootstrap cartridge cluster from existing cluster (#1893)
tarantool/cartridge: rst/cartridge_admin.rst

``http://<instance_hostname>:<instance_http_port>``
(in this example, ``http://localhost:8080``).

-------------------------------------------------------------------------------
Bootstraping from an existing cluster (optional)
-------------------------------------------------------------------------------

> You can bootstrap a cluster from an existing cluster (*original* cluster in the example
>  below) via the argparse option ``TARANTOOL_BOOTSTRAP_FROM`` or ``--bootstrap_from``
>  in the following form: 
>  ``TARANTOOL_BOOTSTRAP_FROM=admin:SECRET-ORIGINAL-CLUSTER-COOKIE@HOST:MASTER_PORT``.

Reviewer: ```suggestion
You can bootstrap a cluster from an existing cluster (*original* cluster in the example
below) via the argparse option ``TARANTOOL_BOOTSTRAP_FROM`` or ``--bootstrap_from``
in the following form: 
``TARANTOOL_BOOTSTRAP_FROM=admin:SECRET-ORIGINAL-CLUSTER-COOKIE@HOST:MASTER_PORT``.
```

                // it's not necessary to bootstrap it from original cluster
                "join_servers": [
                    {
                        "uri": "localhost:13301", // <- different uri
                        "uuid": "cccccccc-cccc-0000-0000-000000000001"
                    }
                ],
                "roles": ["vshard-router", "failover-coordinator"]
            },
            {
                "alias": "storage-original",
                "uuid": "bbbbbbbb-0000-0000-0000-000000000000", // replicaset_uuid is the same as in the original cluster
                // that allows us bootstrap target cluster from original cluster
                "weight": 1,
                "join_servers": [
                    {
                        "uri": "localhost:13302", // <- different uri
                        "uuid": "bbbbbbbb-bbbb-0000-0000-000000000002" // <- different instance_uuid
                    }
                ],
                "roles": ["vshard-storage"]
            }
        ]
    }

-------------------------------------------------------------------------------

> Setup cluster

Reviewer: For consistency, this should be either **Setting up cluster** (most of other headings are in the gerund form "Doing something") or **Cluster setup** (simple noun form).

``http://<instance_hostname>:<instance_http_port>``
(in this example, ``http://localhost:8080``).

(Optional) You can bootstrap cluster from existing cluster (*original* cluster in example)
via argparse option ``TARANTOOL_BOOTSTRAP_FROM`` or ``--bootstrap_from`` in form
``TARANTOOL_BOOTSTRAP_FROM=admin:SECRET-ORIGINAL-CLUSTER-COOKIE@HOST:MASTER_PORT``.
That option should be present on each instance in replicasets of the second cluster.
Make sure that you prepared valid configuration for the second cluster.

> Valid topology should contain **same** replicasets uuids for each replicaset

Reviewer 2: Lets start to descrbe valid configuration. Eg
Valid configuration is:
  - valid topology.yaml (or edit topology)

Reviewer 2: The data will be transfered using iproto replication.

``http://<instance_hostname>:<instance_http_port>``
(in this example, ``http://localhost:8080``).

> (Optional) You can bootstrap cluster from existing cluster (*original* cluster in example)

Reviewer: For better structure, I would add this part under a separate heading. Something like *Bootstrapping from an existing cluster*.

> (Optional) You can bootstrap cluster from existing cluster (*original* cluster in example)
> via argparse option ``TARANTOOL_BOOTSTRAP_FROM`` or ``--bootstrap_from`` in form
> ``TARANTOOL_BOOTSTRAP_FROM=admin:SECRET-ORIGINAL-CLUSTER-COOKIE@HOST:MASTER_PORT``.

Reviewer: ```suggestion
(Optional) You can bootstrap a cluster from an existing cluster (*original* cluster in the example
 below) via the argparse option ``TARANTOOL_BOOTSTRAP_FROM`` or ``--bootstrap_from``
 in the following form: 
 ``TARANTOOL_BOOTSTRAP_FROM=admin:SECRET-ORIGINAL-CLUSTER-COOKIE@HOST:MASTER_PORT``.
```

That option should be present on each instance in replicasets of the target cluster.

> Make sure that you prepared valid configuration for the target cluster.
> Valid topology should contain **same** replicasets uuids for each replicaset
> and instances uuids that **differ** from original cluster.

Reviewer: ```suggestion
Make sure that you've prepared a valid configuration for the target cluster.
A valid topology should contain the **same** *replicaset uuids* for each replicaset
and *instance uuids* that **differ** from original cluster.
```

Several notes:

> - You can bootstrap several replicasets from cluster
>   (for example, data nodes only) instead of full cluster.

Reviewer: ```suggestion
- You can bootstrap specific replicasets from a cluster
  (for example, data nodes only) instead of the whole cluster.
```

- Make sure you aren't trying to load data in target cluster while bootstrapping.
  If you need to hot switch between original and target cluster, make original cluster

>   read-only before bootstraping target cluster.

Reviewer 3: I'm not sure, that this will work. Original cluster must write new members to `_cluster` space, and it will fail to do so, if it is read-only.

>   in original cluster before bootstraping target cluster.

Reviewer: "Make sure you aren't trying to load data" sounds a bit weird: you either try to do something or not.
So, we can say directly something like "Don't load data in the target cluster while bootstrapping" or "... unless bootstrapping is completed". Will this be correct?
